fun/f_sin_clip.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 The Gosl Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fun
+
+import (
+	"math"
+
+	"github.com/cpmech/gosl/chk"
+)
+
+// Sin_clip implements if y < 0: y(t) = a * sin(b*t) + c
+type Sin_clip struct {
```

Comment:
It's better not to use underscores in class names. Maybe a better name would be: CutSin and the key would be "cut-sin"